### PR TITLE
Droth 922 improvement new speed limit data

### PR DESCRIPTION
**Pull request by joanacoelho**

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala [hunk 1]
```diff
@@ -375,20 +392,92 @@ class SpeedLimitsTierekisteriImporter extends LinearAssetTierekisteriImporterOpe
...
+          val newTrAsset = TierekisteriTrafficSignData(newStartAddressSection.roadNumber, newStartAddressSection.roadPartNumber, newStartAddressSection.roadPartNumber, newStartAddressSection.track, newStartAddressSection.startAddressMValue, newStartAddressSection.startAddressMValue, roadSide, TRTrafficSignType.EndUrbanArea, defaultSpeedLimit.toString)
+          createLinearAsset(roadLink, roadAddress, newStartAddressSection, measure, newTrAsset)
+        case _ => None
+      }
+    }
+
+    //Middle Part of The Section Splited
+    if ((trUAAsset.startAddressMValue >= addressSection.startAddressMValue) && (trUAAsset.endAddressMValue <= addressSection.endAddressMValue.head)) {
+      val newMiddleAddressSection = addressSection.copy(startAddressMValue = trUAAsset.startAddressMValue, endAddressMValue = Some(trUAAsset.endAddressMValue))
+      calculateMeasures(roadAddress, newMiddleAddressSection) match {
```
Comment (joanacoelho) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala]: Maybe this part of the code can be put in some method in order to be use in the next if's

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriDataImporter.scala [hunk 2]
```diff
@@ -23,6 +23,7 @@ class TierekisteriDataImporter(vvhClient: VVHClient, oracleLinearAssetDao: Oracl
   val damagedByThawAssetId = 130
   val europeanRoadAssetId = 260
   val speedLimitAssetId = 20
+  val SpeedLimitStateAssetId = 310
```
Comment (oskarmalmstrom) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriDataImporter.scala]: first letter lower case

**Comment by marquesfm** [comment]
ok for me

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala [hunk 3]
```diff
@@ -375,20 +393,86 @@ class SpeedLimitsTierekisteriImporter extends LinearAssetTierekisteriImporterOpe
...
+    }
+
+    trUrbanAreaAssets.filter(_.startRoadPartNumber == addressSection.roadPartNumber).foreach(
+      trUA =>
+        if ((trUA.startAddressMValue <= addressSection.startAddressMValue) && (trUA.endAddressMValue >= addressSection.endAddressMValue.head)) {
+          return createSpeedLimitByUrbanArea(trUA.assetValue, addressSection)
+        } else {
+          createMultiSpeedLimitInAddressSection(trUA, roadLink, addressSection, roadAddress, roadSide)
+          return None
+        }
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala]: Why do we use the returns!?!?

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala [hunk 4]
```diff
@@ -375,20 +393,86 @@ class SpeedLimitsTierekisteriImporter extends LinearAssetTierekisteriImporterOpe
...
+      val newMiddleAddressSection = addressSection.copy(startAddressMValue = trUAAsset.startAddressMValue, endAddressMValue = Some(trUAAsset.endAddressMValue))
+      createSectionPart(roadAddress, newMiddleAddressSection, TRTrafficSignType.UrbanArea, urbanAreaSpeedLimit.toString)
+    }
+
+    //Final Part of The Section Splited
+    if ((trUAAsset.endAddressMValue < addressSection.endAddressMValue.head) && (verifyDefaultTolerance(trUAAsset.endAddressMValue, addressSection.endAddressMValue.head)) && (trUAAsset.endAddressMValue >= addressSection.startAddressMValue)) {
+      val newFinalAddressSection = addressSection.copy(startAddressMValue = trUAAsset.endAddressMValue)
+      createSectionPart(roadAddress, newFinalAddressSection, TRTrafficSignType.EndUrbanArea, defaultSpeedLimit.toString)
+    }
+  }
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TierekisteriImporterOperations.scala]: This logic should get in account the "9"?

**Pull request merged**